src/pykeen/trackers/tensorboard.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+"""An adapter for TensorBoard."""
+
+import datetime
+import pathlib
+from typing import Any, Dict, Mapping, Optional, TYPE_CHECKING, Type
+
+from .base import ResultTracker
+from ..constants import PYKEEN_LOGS
+from ..utils import flatten_dictionary
+
+if TYPE_CHECKING:
+    import torch.utils.tensorboard
+
+__all__ = [
+    'TensorBoardResultTracker',
+]
+
+
+class TensorBoardResultTracker(ResultTracker):
+    """A tracker for TensorBoard."""
+
+    #: The class that's used to instantiate a summarywriter
+    SummaryWriter: Type['torch.utils.tensorboard.SummaryWriter']
```

Comment:
should this be `summary_writer_cls`?

Reply:
I am actually not sure, this was a change Charlie added.

Reply:
@cthoyt 🙂

Reply:
yes this should also be changed

Reply:
https://github.com/sbonner0/pykeen/commit/dec8030c760ae077967201d29b8e0e215f724462